app/src/main/java/com/example/tasky/feature_agenda/data/worker/EventWorker.kt

```diff
@@ -0,0 +1,203 @@
+                    from = event.from.toUtcTimestamp(),
+                    to = event.to.toUtcTimestamp(),
+                    remindAt = event.remindAt.toUtcTimestamp(),
+                    attendeeIds = event.attendees.map { it.userId }
+                ),
+                photoList
+            )
+            Resource.Success()
+        } catch(e: HttpException) {
+            Resource.Error("Invalid Response")
+        } catch(e: IOException) {
+            Resource.Error("Couldn't reach server")
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    private suspend fun getEvent(eventId: String): Resource<AgendaItem.Event> {
+
+        return try {
+            val response = api.getEvent(eventId)
+            val event = response.body()?.toEvent() ?: return Resource.Error("No such event found!")
+
+            val eventEntity = event.toEventEntity()
+            db.eventDao.upsertEvent(eventEntity)
+            val newEvent = db.eventDao.getEventWithAttendees(eventId.toInt()) ?: return Resource.Error("No such event found!")
+
+            Resource.Success(newEvent.toEvent())
+        } catch(e: HttpException) {
+            Resource.Error("Invalid Response")
+        } catch(e: IOException) {
+            Resource.Error("Couldn't reach server")
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    private suspend fun updateEvent(event: AgendaItem.Event, deletedPhotos: List<Photo> = emptyList()): Resource<Unit> {
+
+        val photoList = event.photos.map { photo ->
+            val file = File(photo.url)
+            val requestFile: RequestBody = file.asRequestBody("image/*".toMediaTypeOrNull())
+            MultipartBody.Part.createFormData("photos", file.name, requestFile)
+        }
+
+        val userId = userPrefs.getAuthenticatedUser()?.userId ?: return Resource.Error("User is not logged in!")
+
+        return try {
+            api.updateEvent(
+                UpdateEventRequest(
+                    id = event.eventId,
+                    title = event.title,
+                    description = event.description ?: "",
+                    from = event.from.toUtcTimestamp(),
+                    to = event.to.toUtcTimestamp(),
+                    remindAt = event.remindAt.toUtcTimestamp(),
+                    attendeeIds = event.attendees.map { it.userId },
+                    deletedPhotoKeys = deletedPhotos.map { it.key },
+                    isGoing = event.attendees.any{ it.userId == userId }
+                ),
+                photoList
+            )
+
+            Resource.Success()
+        } catch(e: HttpException) {
+            Resource.Error("Invalid Response")
+        } catch(e: IOException) {
+            Resource.Error("Couldn't reach server")
+        }
+    }
+
+    private suspend fun deleteEventAndAttendees(eventId: String): Resource<Unit> {
+        return try {
+            api.deleteEvent(eventId)
+            api.deleteAttendee(eventId)
```

Comment:
-`deleteAttendee` should be used if you're an attendee of an event and you delete it from your agenda
-`deleteEvent` is used if you're the event owner and you delete the event. Then all attendees will be deleted by the API